server/src/main/java/org/elasticsearch/common/blobstore/EndpointStats.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.common.blobstore;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * The number of calls to an endpoint
+ *
+ * @param operations The number of calls
+ * @param requests The number of calls (including retries)
+ */
+public record EndpointStats(long operations, long requests) implements Writeable, ToXContentObject {
+
+    public static EndpointStats ZERO = new EndpointStats(0, 0);
+
+    public EndpointStats(StreamInput in) throws IOException {
+        this(in.readLong(), in.readLong());
+    }
+
+    public EndpointStats {
+        assert operations >= 0 && requests >= 0;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeLong(operations);
+        out.writeLong(requests);
+    }
+
+    public EndpointStats add(EndpointStats other) {
+        return new EndpointStats(Math.addExact(operations, other.operations), Math.addExact(requests, other.requests));
+    }
+
+    public boolean isZero() {
+        return operations == 0 && requests == 0;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof EndpointStats other) {
+            return requests == other.requests && operations == other.operations;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(operations, requests);
+    }
```

Comment:
Do we need these given it is a record?

Reply:
Addressed in 01439793eb4dd575e3311182906c75fb896846ff